go/pkg/apis/enricher/dotnet_enricher.go

```diff
@@ -35,9 +35,31 @@ func (j DotNetEnricher) DoEnrichLanguage(language *model.Language, files *[]stri
 	}
 }
 
-func (j DotNetEnricher) DoEnrichComponent(component *model.Component) {
+func (j DotNetEnricher) DoEnrichComponent(component *model.Component, settings model.DetectionSettings) {
 	projectName := GetDefaultProjectName(component.Path)
 	component.Name = projectName
+
+	for _, algorithm := range settings.PortDetectionStrategy {
+		ports := []int{}
+		switch algorithm {
+		case model.DockerFile:
```

Comment:
I think all strategies should be used. ex for Quarkus and deploy we can have a default port in source and port in dockerfile.

Reply:
This is the dotnet _enricher and dot net languages do not have any source detection for the moment. This is why only those 2 algos are in the switch.

For the second point i wonder what's the point of performing all detections.
If you are deploying the app then the port in dockerfile should have priority over the source. Probably it would be the same otherwise there is some issue. If you work locally the one in the source code should be used. By defining the strategy order you should take what you need, no?
The only case it can be useful is if the port in the dockerfile is wrong. E.g it tells to open port 9090 and the source code listen to 9091. in that case it's a user fault.
By executing all strategies we could end up opening more ports than those actually required if some file is dirty.